doc/cli_subprocess_guidelines.md

```diff
@@ -0,0 +1,112 @@
+# Azure CLI Subprocess Guidelines
+
+In certain CLI modules, there are scenarios that need to call a subprocess to run commands outside CLI, like getting kubectl info in aks, or deployment setup in mysql using `git` and `gh`, through python built-in [subprocess](https://docs.python.org/3/library/subprocess.html) module. Despite its simplicity and versatility, ensuring the security of applying it can be challenging in different platforms under different circumstance, and it is error-prone for developers to neglect security best practices during development.
+
+
+## Insecure Use Of Commands Under Subprocess Module
+
+
+### Dynamic command construction under risk
+
+Assume a script that needs to read in `user_input` from users to execute a `git` related command and runs it through subprocess using shell. In regular cases, users would add a valid git command, like `git --version` or something else, but if undesired input gets logged into this system, like `--version;echo aa`, the appended `;echo aa` would be executed by `subprocess` too, like below: 
+
+```python
+import subprocess
+user_input = input("input a git command to run: ")
+cmd = f"git {user_input}"
+subprocess.run(cmd, shell=True)
+```
+
+```python
+input a git command to run: --version;echo aa
+git version 2.34.1
+aa
+```
```

Comment:
This is not Python code.

Reply:
right, it is a mock for the console input and output of running python code above